addons/event_sale/models/sale_order.py

```diff
@@ -110,28 +111,41 @@ def _update_registrations(self, confirm=True, cancel_to_draft=False, registratio
             RegistrationSudo.create(registrations_vals)
         return True
 
-    @api.onchange('product_id')
-    def _onchange_product_id(self):
-        # We reset the event when keeping it would lead to an inconstitent state.
-        # We need to do it this way because the only relation between the product and the event is through the corresponding tickets.
-        if self.event_id and (not self.product_id or self.product_id.id not in self.event_id.mapped('event_ticket_ids.product_id.id')):
-            self.event_id = None
-
-    @api.onchange('event_id')
-    def _onchange_event_id(self):
-        # We reset the ticket when keeping it would lead to an inconstitent state.
-        if self.event_ticket_id and (not self.event_id or self.event_id != self.event_ticket_id.event_id):
-            self.event_ticket_id = None
-
-    @api.onchange('product_uom', 'product_uom_qty')
-    def product_uom_change(self):
-        if not self.event_ticket_id:
-            super(SaleOrderLine, self).product_uom_change()
-
-    @api.onchange('event_ticket_id')
-    def _onchange_event_ticket_id(self):
+    @api.depends('product_id')
+    def _compute_event_id(self):
+        event_lines = self.filtered(lambda line: line.product_id and line.product_id.detailed_type == 'event')
+        (self - event_lines).event_id = False
+        for line in event_lines:
+            if line.product_id not in line.event_id.event_ticket_ids.product_id:
+                line.event_id = False
+
+    @api.depends('event_id')
+    def _compute_event_ticket_id(self):
+        event_lines = self.filtered('event_id')
+        (self - event_lines).event_ticket_id = False
+        for line in event_lines:
+            if line.event_id != line.event_ticket_id.event_id:
+                line.event_ticket_id = False
+
+    @api.depends('event_ticket_id')
+    def _compute_price_unit(self):
+        """Do not update the price on qty/uom change"""
+        event_lines = self.filtered('event_ticket_id')
+        super(SaleOrderLine, self-event_lines)._compute_price_unit()
+        for line in event_lines:
+            if not line.product_id or line._origin.product_id != line.product_id:
+                super(SaleOrderLine, line)._compute_price_unit()
+
+    @api.depends('event_ticket_id')
+    def _compute_discount(self):
+        """Do not compute the discount on event lines, it's always included in the price."""
+        event_lines = self.filtered('event_ticket_id')
+        super(SaleOrderLine, self-event_lines)._compute_discount()
+
+    @api.depends('event_ticket_id')
+    def _compute_name(self):
```

Comment:
Same here, where is the logic of the name computation ? Is it the ``_multiline_description_sale`` ? In any case a comment would be welcomed, otherwise people might remove that code that looks unnecessary.